Utoipa 5.0.0-rc: Name conflict for functions in different modules

### Description:
When using Utoipa 5.0.0-rc.0, I encountered a compilation error when trying to generate OpenAPI documentation for two functions with the same name in different modules.

### Version Information:
- Utoipa: 5.0.0-rc.0
- Rust: cargo 1.81.0 (2dbb1af80 2024-08-20)
- OS: MacOS 15.0

### Expected Behavior:
The OpenAPI documentation should be generated successfully for all routes, regardless of function names, as long as they are in different modules.

### Actual Behavior:
The compilation fails with the following error:
```
$ cargo clippy
    Checking utoipa-bug v0.1.0 (/Users/robbie/Documents/projects/rust/utoipa-bug)
error[E0428]: the name `___path_example` is defined multiple times
  --> src/main.rs:34:10
   |
34 | #[derive(OpenApi)]
   |          ^^^^^^^ `___path_example` redefined here
   |
   = note: `___path_example` must be defined only once in the type namespace of this block
   = note: this error originates in the derive macro `OpenApi` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0119]: conflicting implementations of trait `utoipa::__dev::PathConfig` for type `<ApiDoc as utoipa::OpenApi>::openapi::___path_example`
  --> src/main.rs:34:10
   |
34 | #[derive(OpenApi)]
   |          ^^^^^^^
   |          |
   |          first implementation here
   |          conflicting implementation for `<ApiDoc as utoipa::OpenApi>::openapi::___path_example`
   |
   = note: this error originates in the derive macro `OpenApi` (in Nightly builds, run with -Z macro-backtrace for more info)

Some errors have detailed explanations: E0119, E0428.
For more information about an error, try `rustc --explain E0119`.
error: could not compile `utoipa-bug` (bin "utoipa-bug") due to 2 previous errors
```

### Minimal Reproducible Example:
```rust
// src/main.rs
use axum::{routing::get, Router};
use utoipa::OpenApi;
use utoipa_scalar::{Scalar, Servable};

mod foo {
    /// Foo Example
    #[utoipa::path(get, path = "/foo", operation_id = "foo")]
    pub async fn example() -> &'static str {
        "ok"
    }
}

mod bar {
    /// Bar Example
    #[utoipa::path(get, path = "/bar", operation_id = "bar")]
    pub async fn example() -> &'static str {
        "ok"
    }
}

#[derive(OpenApi)]
#[openapi(paths(foo::example, bar::example))]
struct ApiDoc;

#[tokio::main]
async fn main() {
    let app = Router::new()
        .merge(Scalar::with_url("/docs", ApiDoc::openapi()))
        .route("/foo", get(foo::example))
        .route("/bar", get(bar::example));

    let listener = tokio::net::TcpListener::bind("127.0.0.1:3000")
        .await
        .unwrap();

    axum::serve(listener, app).await.unwrap();
}
```

```toml
# Cargo.toml
[package]
name = "utoipa-bug"
version = "0.1.0"
edition = "2021"

[dependencies]
axum = "0.7"
tokio = { version = "1", features = ["macros", "rt-multi-thread"] }
utoipa = "5.0.0-rc.0"
utoipa-scalar = { version = "0.2.0-rc.0", features = ["axum"] }
```

I believe this is a bug as everything works perfectly fine in Utoipa 4 and I believe that having multiple functions share names but be isolated by module (e.g: `user::create`, `diagram::create`) is valid.

I'm happy to provide any additional information or context that might be helpful in resolving this issue. Thank you for your work on Utoipa!

_Edit:_ With the release of `rc.0` I've updated and tested, this is still an issue in `rc.0`

--- a/.github/actions/gitlog/gitlog.sh
+++ b/.github/actions/gitlog/gitlog.sh
@@ -70,11 +70,15 @@ mapfile -t log_lines < <(git log --pretty=format:'(%h) %s' $ancestry_path "$comm
 function is_crate_related {
     commit="$1"
     changes="$(git diff --name-only "$commit"~ "$commit" | awk -F / '{print $1}' | xargs)"
+    IFS=" " read -r -a change_dirs <<<"$changes"
 
     is_related=false
-    if [[ "$changes" == *"$crate"* ]]; then
-        is_related=true
-    fi
+    for change in "${change_dirs[@]}"; do
+        if [[ "$change" == "$crate" ]]; then
+            is_related=true
+            break
+        fi
+    done
 
     echo $is_related
 }

--- a/utoipa-gen/CHANGELOG.md
+++ b/utoipa-gen/CHANGELOG.md
@@ -24,6 +24,7 @@
 
 ### Fixed
 
+* Fix nest path config struct name (https://github.com/juhaku/utoipa/pull/1081)
 * Fix `as` attribute path format (https://github.com/juhaku/utoipa/pull/1080)
 * Fix allow response `content_type` without schema (https://github.com/juhaku/utoipa/pull/1073)
 * Fix negative value parsing on schema attributes (https://github.com/juhaku/utoipa/pull/1031)

--- a/utoipa-gen/src/openapi.rs
+++ b/utoipa-gen/src/openapi.rs
@@ -691,9 +691,14 @@ fn impl_paths(handler_paths: Option<&Punctuated<ExprPath, Comma>>) -> Paths {
         .flatten()
         .map(|handler| {
             let segments = handler.path.segments.iter().collect::<Vec<_>>();
+            let handler_config_name = segments
+                .iter()
+                .map(|segment| segment.ident.to_string())
+                .collect::<Vec<_>>()
+                .join("_");
             let handler_fn = &segments.last().unwrap().ident;
             let handler_ident = path::format_path_ident(Cow::Borrowed(handler_fn));
-            let handler_ident_nested = format_ident!("_{}", handler_ident.as_ref());
+            let handler_ident_config = format_ident!("{}_config", handler_config_name);
 
             let tag = segments
                 .iter()
@@ -713,7 +718,7 @@ fn impl_paths(handler_paths: Option<&Punctuated<ExprPath, Comma>>) -> Paths {
                 .join("::"),
             )
             .unwrap();
-            (usage, tag, handler_ident_nested)
+            (usage, tag, handler_ident_config)
         })
         .collect::<Vec<_>>();
 
@@ -749,12 +754,15 @@ fn impl_paths(handler_paths: Option<&Punctuated<ExprPath, Comma>>) -> Paths {
         quote! { #handlers_impls utoipa::openapi::path::PathsBuilder::new() },
         |mut paths, handler| {
             let segments = handler.path.segments.iter().collect::<Vec<_>>();
-            let handler_fn = &segments.last().unwrap().ident;
-            let handler_ident_nested =
-                format_ident!("_{}", path::format_path_ident(Cow::Borrowed(handler_fn)));
+            let handler_config_name = segments
+                .iter()
+                .map(|segment| segment.ident.to_string())
+                .collect::<Vec<_>>()
+                .join("_");
+            let handler_ident_config = format_ident!("{}_config", handler_config_name);
 
             paths.extend(quote! {
-                .path_from::<#handler_ident_nested>()
+                .path_from::<#handler_ident_config>()
             });
 
             paths

